### ISSUE TITLE
Cannot view all projects when deployed in a cluster.

### ISSUE THREAD

**Issue by Sajotim** [sentry-kubernetes/charts#251]
When I click Project in the sidebar, Sentry prompts me 
**Unable to fetch all project stats**
Click Projects in Organization Settings and it will crash directly.
I checked the log of the container as follows

```
2020-11-24 00:35:43 | QueryExecutionError: [60] DB::Exception: Table default.sentry_dist doesn't exist.. Stack trace:
-- | --
  |   | 2020-11-24 00:35:43 | error["message"]
  |   | 2020-11-24 00:35:43 | File "/usr/local/lib/python2.7/site-packages/sentry/utils/snuba.py", line 712, in bulk_raw_query
  |   | 2020-11-24 00:35:43 | return bulk_raw_query([snuba_params], referrer=referrer)[0]
  |   | 2020-11-24 00:35:43 | File "/usr/local/lib/python2.7/site-packages/sentry/utils/snuba.py", line 633, in raw_query
  |   | 2020-11-24 00:35:43 | **kwargs
  |   | 2020-11-24 00:35:43 | File "/usr/local/lib/python2.7/site-packages/sentry/utils/snuba.py", line 755, in query
  |   | 2020-11-24 00:35:43 | is_grouprelease=(model == TSDBModel.frequent_releases_by_group),
  |   | 2020-11-24 00:35:43 | File "/usr/local/lib/python2.7/site-packages/sentry/tsdb/snuba.py", line 288, in get_data
  |   | 2020-11-24 00:35:43 | conditions=conditions,
  |   | 2020-11-24 00:35:43 | File "/usr/local/lib/python2.7/site-packages/sentry/tsdb/snuba.py", line 363, in get_range
  |   | 2020-11-24 00:35:43 | return getattr(self.backends[backend], key)(*a, **kw)
  |   | 2020-11-24 00:35:43 | File "/usr/local/lib/python2.7/site-packages/sentry/tsdb/redissnuba.py", line 87, in method
  |   | 2020-11-24 00:35:43 | context[key] = (lambda f: lambda *a, **k: getattr(self, f)(*a, **k))(key)
  |   | 2020-11-24 00:35:43 | File "/usr/local/lib/python2.7/site-packages/sentry/utils/services.py", line 105, in <lambda>
  |   | 2020-11-24 00:35:43 | model=stat_model, keys=keys, **self._parse_args(request, **query_kwargs)
  |   | 2020-11-24 00:35:43 | File "/usr/local/lib/python2.7/site-packages/sentry/api/endpoints/organization_stats.py", line 94, in get
  |   | 2020-11-24 00:35:43 | response = handler(request, *args, **kwargs)
  |   | 2020-11-24 00:35:43 | File "/usr/local/lib/python2.7/site-packages/sentry/api/base.py", line 237, in dispatch
  |   | 2020-11-24 00:35:43 | self.raise_uncaught_exception(exc)
  |   | 2020-11-24 00:35:43 | File "/usr/local/lib/python2.7/site-packages/rest_framework/views.py", line 449, in handle_exception
  |   | 2020-11-24 00:35:43 | response = super(Endpoint, self).handle_exception(exc)
  |   | 2020-11-24 00:35:43 | File "/usr/local/lib/python2.7/site-packages/sentry/api/base.py", line 124, in handle_exception
  |   | 2020-11-24 00:35:43 | Traceback (most recent call last):
  |   | 2020-11-24 00:35:43 | 202.202.43.125 - - [23/Nov/2020:16:35:43 +0000] "GET /api/0/organizations/sentry/projects/?query=&per_page=50 HTTP/1.0" 200 2669 "https://sentry.redrock.team/settings/sentry/projects/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.1 Safari/605.1.15"
  |   | 2020-11-24 00:35:43 | 202.202.43.125 - - [23/Nov/2020:16:35:43 +0000] "GET /_static/1606148990/sentry/dist/OrganizationProjects.js HTTP/1.0" 200 3661 "https://sentry.redrock.team/settings/sentry/projects/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.1 Safari/605.1.15"
  |   | 2020-11-24 00:35:39 | 202.202.43.125 - - [23/Nov/2020:16:35:39 +0000] "GET /_health/ HTTP/1.1" 200 203 "-" "kube-probe/1.19"
  |   | 2020-11-24 00:35:37 | 16:35:37 [ERROR] sentry_sdk.errors: Internal error in sentry_sdk
  |   | 2020-11-24 00:35:37 | MaxRetryError: HTTPConnectionPool(host='sentry.redrock.team', port=80): Max retries exceeded with url: /api/1/store/ (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7ff8008457d0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',))
  |   | 2020-11-24 00:35:37 | raise MaxRetryError(_pool, url, error or ResponseError(cause))
```

I found a person who had a similar problem with me, but he deployed using docker-compose
[Sentry 10 error after updating Docker images](https://forum.sentry.io/t/sentry-10-error-after-updating-docker-images/8880/4)
BYK says that executing docker-compose run --rm snuba-api migrate can solve it
I tried to enter the snuba container and executed snuba migrations migrate, but it was unsuccessful. 
It prompted **Cannot run migrations for multi node clusters.**
Has anyone had the same problem when deploying to a kubernetes cluster? Hope to solve this problem

**Comment by fergusean** [comment]
@Sajotim did you get this resolved? I'm experiencing the same now after migrating a local Docker install to a Kubernetes install.

**Comment by ZacharyJoswick** [comment]
I had a different issue that was related to missing tables after pods were moved between nodes, but I resolved it by doing the following:

1. Kubectl exec into the sentry snuba api pod
2. Run `export CLICKHOUSE_SINGLE_NODE="true"`
3. Then run the following:
  ```
 for ((i=0;i<3;i++)); do
    export CLICKHOUSE_HOST=sentry-clickhouse-$i.sentry-clickhouse-headless;
    snuba migrate;
    export CLICKHOUSE_HOST=sentry-clickhouse-replica-$i.sentry-clickhouse-replica-headless;
    snuba migrate;
done
  ```

This is roughly the same thing the sentry-snuba-migrate job does in the helm chart.

Not 100% sure if it will solve your issues, but its worth a shot.
